src/Polly.Core.Benchmarks/README.md

```diff
@@ -14,14 +14,42 @@ LaunchCount=2  WarmupCount=10
+| ExecutePipeline_V7 |          2 |   164.39 ns |  5.294 ns |  7.592 ns |  1.00 |    0.00 | 0.0658 |     552 B |        1.00 |
+| ExecutePipeline_V8 |          2 |   126.74 ns |  1.198 ns |  1.755 ns |  0.77 |    0.04 | 0.0181 |     152 B |        0.28 |
 |                    |            |             |           |           |       |         |        |           |             |
-| ExecutePipeline_V7 |          5 |   531.04 ns |  4.728 ns |  6.930 ns |  1.00 |    0.00 | 0.1545 |    1296 B |        1.00 |
-| ExecutePipeline_V8 |          5 |   245.50 ns |  2.344 ns |  3.509 ns |  0.46 |    0.01 | 0.0048 |      40 B |        0.03 |
+| ExecutePipeline_V7 |          5 |   540.07 ns | 16.941 ns | 24.832 ns |  1.00 |    0.00 | 0.1545 |    1296 B |        1.00 |
+| ExecutePipeline_V8 |          5 |   257.13 ns |  2.748 ns |  4.114 ns |  0.48 |    0.03 | 0.0181 |     152 B |        0.12 |
 |                    |            |             |           |           |       |         |        |           |             |
-| ExecutePipeline_V7 |         10 | 1,128.82 ns | 10.838 ns | 15.886 ns |  1.00 |    0.00 | 0.3014 |    2536 B |        1.00 |
-| ExecutePipeline_V8 |         10 |   449.31 ns |  2.926 ns |  4.379 ns |  0.40 |    0.01 | 0.0048 |      40 B |        0.02 |
\ No newline at end of file
+| ExecutePipeline_V7 |         10 | 1,111.72 ns | 16.405 ns | 23.527 ns |  1.00 |    0.00 | 0.3014 |    2536 B |        1.00 |
+| ExecutePipeline_V8 |         10 |   467.93 ns |  6.546 ns |  9.388 ns |  0.42 |    0.01 | 0.0181 |     152 B |        0.06 |
+
+## TIMEOUT
+
+|            Method |     Mean |   Error |   StdDev | Ratio | RatioSD |   Gen0 | Allocated | Alloc Ratio |
+|------------------ |---------:|--------:|---------:|------:|--------:|-------:|----------:|------------:|
+| ExecuteTimeout_V7 | 287.1 ns | 9.20 ns | 12.59 ns |  1.00 |    0.00 | 0.0868 |     728 B |        1.00 |
+| ExecuteTimeout_V8 | 272.9 ns | 3.16 ns |  4.54 ns |  0.95 |    0.04 | 0.0439 |     368 B |        0.51 |
+
+## RETRY
+
+|          Method |     Mean |   Error |  StdDev | Ratio | RatioSD |   Gen0 | Allocated | Alloc Ratio |
+|---------------- |---------:|--------:|--------:|------:|--------:|-------:|----------:|------------:|
+| ExecuteRetry_V7 | 162.5 ns | 2.33 ns | 3.34 ns |  1.00 |    0.00 | 0.0687 |     576 B |        1.00 |
+| ExecuteRetry_V8 | 152.3 ns | 1.31 ns | 1.93 ns |  0.94 |    0.02 | 0.0181 |     152 B |        0.26 |
+
+## RATE LIMITER
+
+|                Method |     Mean |   Error |  StdDev | Ratio | RatioSD |   Gen0 | Allocated | Alloc Ratio |
+|---------------------- |---------:|--------:|--------:|------:|--------:|-------:|----------:|------------:|
+| ExecuteRateLimiter_V7 | 173.8 ns | 2.33 ns | 3.48 ns |  1.00 |    0.00 | 0.0448 |     376 B |        1.00 |
+| ExecuteRateLimiter_V8 | 207.9 ns | 2.06 ns | 2.89 ns |  1.19 |    0.03 | 0.0229 |     192 B |        0.51 |
```

Comment:
Interesting that this one is slower.

Reply:
Yup surprised me too. It also allocates on every call (around 40 bytes, the remaining 152 bytes will be addressed by pooling the `ResilienceContext`).  

Might be something that could be addressed with .NET folks. 

cc @juraj-blazek @geeknoid